Automattic/wp-calypso
# Checkout: Hide postalCode field in TaxFields if country does not support it

#### Changes proposed in this Pull Request

There are countries that do not have a postal code, and for those we should not require one in checkout. This has already been done in https://github.com/Automattic/wp-calypso/pull/53522 and https://github.com/Automattic/wp-calypso/pull/54327 for domain fields, but the "tax" fields (the contact fields shown when there is not a domain or G Suite in the cart) were not changed. This PR hides the field for the tax form as well if the country does not support postal codes.

Because the decision to display the postal code field depends on the country selection, this PR also reverses the order of the fields so that instead of POSTAL, COUNTRY, it's now COUNTRY, POSTAL.

Requires D68482-code

#### Testing instructions

First apply D68482-code and sandbox the store.

- Visit checkout with a non-domain product in your cart (eg: a plan).
- Verify that the country select list is displayed and looks ok.
- Select a country with a postal code (most of them) and verify that the form now displays the postal code field as well.
- Verify that you cannot complete the step without filling in the postal code, but that once it is filled (and valid), the "Continue" button will move you to the next step.

Next...

- Repeat the above steps but select a country code that does not have postal codes, like Curaçao.
- Verify that the postal code field is not displayed.
- Verify that you can complete the contact step without filling in the field.
